webhook_to_fedora_messaging/endpoints/service.py

```diff
@@ -1,166 +1,182 @@
-from .util import validate_request
-
-
-service_endpoint = Blueprint("service_endpoint", __name__, url_prefix="/service")
-
-
-@service_endpoint.route("/", methods=["POST"])
-@validate_request(["username", "type", "desc", "name"])
-def create_service():
+import logging
+from uuid import uuid4
+
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy import select
+from sqlalchemy.exc import IntegrityError, NoResultFound
+from sqlalchemy.ext.asyncio import AsyncSession
+from starlette.status import (
+    HTTP_200_OK,
+    HTTP_201_CREATED,
+    HTTP_202_ACCEPTED,
+    HTTP_409_CONFLICT,
+    HTTP_422_UNPROCESSABLE_ENTITY,
+)
+
+from webhook_to_fedora_messaging.auth import user_factory
+from webhook_to_fedora_messaging.database import get_session
+from webhook_to_fedora_messaging.endpoints.models.service import (
+    ServiceExternal,
+    ServiceManyResult,
+    ServiceRequest,
+    ServiceResult,
+    ServiceUpdate,
+)
+from webhook_to_fedora_messaging.endpoints.util import authorized_service_from_uuid
+from webhook_to_fedora_messaging.models import Service, User
+
+
+logger = logging.getLogger(__name__)
+router = APIRouter(prefix="/services")
+
+
+@router.post(
+    "",
+    status_code=HTTP_201_CREATED,
+    response_model=ServiceResult,
+    tags=["services"]
+)
+async def create_service(
+    body: ServiceRequest,
+    session: AsyncSession = Depends(get_session),  # noqa : B008
+    user: User = Depends(user_factory()),  # noqa : B008
+):
     """
-    Used for creating a new service by sending a post request to /service/ path.
+    Create a service with the requested attributes
     """
-    user = db.session.query(User).filter(User.username == request.json["username"]).first()
-    if user is None:
-        return {"message": "Not Found"}, 404
-
-    service, is_created = get_or_create(
-        db.session,
-        Service,
-        name=request.json["name"],
-        type=request.json["type"],
-        desc=request.json["desc"],
-        user_id=user.id,
+    made_service = Service(
+        name=body.data.name,
+        uuid=uuid4().hex[0:8],
+        type=body.data.type,
+        desc=body.data.desc,
+        user_id=user.id
     )
-    if not is_created:
-        return abort(409, {"message": "Service Already Exists"})
-    else:
-        db.session.commit()
-        return {"message": "Created", "uuid": service.uuid, "token": service.token}, 201
-
-
-@service_endpoint.route("/search", methods=["GET"])
-@validate_request
-def list_services():
+    session.add(made_service)
+    try:
+        await session.flush()
+    except IntegrityError as expt:
+        logger.exception("Uniquness constraint failed - Please try again")
+        raise HTTPException(
+            HTTP_409_CONFLICT,
+            "Uniquness constraint failed - Please try again"
+        ) from expt
+    return {
+        "data": ServiceExternal.model_validate(made_service).model_dump()
+    }
+
+
+@router.get(
+    "/list",
```

Comment:
Getting a list of service is usually done by making a GET request to the root endpoint, ie `GET /services/` rather than `GET /services/list`.

```suggestion
    "",
```